### ISSUE TITLE
Instagram story share not working on Android

### ISSUE THREAD

**Issue by sadatd** [appinioGmbH/flutter_packages#131]
appinia_social_share: 0.2.4
Android version:  13
Flutter version: 3.11.0-0.1.pre
Dart version 3.1.0
When I try to share to instagram story, I get redirected to instagram, but Instagram shows error "Something went wrong. Please try again.". But return message from calling `_appinioShare.shareToInstagramStory` is `SUCCESS`.
It works fine on iOS though.
Here is screenshot of instagram error:
<img width=400 src="https://github.com/appinioGmbH/flutter_packages/assets/15090688/438ad2df-5b81-446b-a0cc-de9d01af18a1"/>

**Comment by sadatd** [comment]
A temporary solution for me was to use `backgroundImage` instead of `stickerImage`, like following: 
```
await _appinioShare.shareToInstagramStory(
      'facebookId',
      // stickerImage: filePath, // TODO: uncomment when instagram story is fixed
      backgroundImage: filePath,
    ); 
```

**Comment by jakobheb** [comment]
What's your settings in AndroidManifest, because in my project it's not even opening Instagram

**Comment by sadatd** [comment]
> What's your settings in AndroidManifest, because in my project it's not even opening Instagram

It is nothing different than settings mentioned in appinio_social_share's 'readme' section. You can share your error here, so somebody can maybe help with it.

**Comment by jakobheb** [comment]
Problem is that I'm not receiving any error, but neither a success message. iOS is working fine, but ShareToInstagramStory and Feed are not opening Instagram.

**Comment by jakobheb** [comment]
`<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:tools="http://schemas.android.com/tools"
    package="com.app.app">

    <queries>
        
        <package android:name="com.zhiliaoapp.musically" />
        <package android:name="com.facebook.katana" />
        <package android:name="com.facebook.orca" />
        <package android:name="org.telegram.messenger" />
        <package android:name="com.whatsapp" />
        <package android:name="com.twitter.android" />
        <package android:name="com.instagram.android" />


        <provider android:authorities="com.facebook.katana.provider.PlatformProvider" /> 
        <provider android:authorities="com.facebook.orca.provider.PlatformProvider" /> 
    </queries>

    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
    <uses-permission android:name="android.permission.MANAGE_EXTERNAL_STORAGE" />
    <uses-permission android:name="android.permission.INTERNET"/>

    <application
        android:label="Gainback"
        android:name="${applicationName}"
        android:icon="@mipmap/ic_launcher">

  

        <provider
            android:name="androidx.core.content.FileProvider"
            android:authorities="${applicationId}.provider"
            android:exported="false"
            android:grantUriPermissions="true">
            <meta-data
                android:name="android.support.FILE_PROVIDER_PATHS"
                android:resource="@xml/provider_paths_app" />

        </provider>

        <provider
            android:name="com.facebook.FacebookContentProvider"
            android:authorities="com.facebook.app.FacebookContentProvider1234"
            android:exported="true" />
        <meta-data
            android:name="com.facebook.sdk.ApplicationId"
            android:value="@string/facebook_app_id" />
        <meta-data
            android:name="com.facebook.sdk.ClientToken"
            android:value="@string/facebook_client_token" />

        <activity
            android:name=".MainActivity"
            android:exported="true"
            android:launchMode="singleTop"
            android:theme="@style/LaunchTheme"
            android:keepScreenOn="true"
            android:configChanges="orientation|keyboardHidden|keyboard|screenSize|smallestScreenSize|locale|layoutDirection|fontScale|screenLayout|density|uiMode"
            android:hardwareAccelerated="true"
            android:windowSoftInputMode="adjustResize">
            
            <meta-data
                android:name="io.flutter.embedding.android.NormalTheme"
                android:resource="@style/NormalTheme"
                />
            <intent-filter>
                <action android:name="android.intent.action.MAIN"/>
                <category android:name="android.intent.category.LAUNCHER"/>
            </intent-filter>
        </activity>
        
        <meta-data
            android:name="flutterEmbedding"
            android:value="2" />
    </application>
</manifest>`

**Comment by sadatd** [comment]
Hey @jakobheb, it looks okay to me. Just one thing to note: please check the location of your `provider_paths_app.xml`; it should be inside `android/app/src/main/res/xml/`, even though the documentation states that it should be inside `android/app/src/main/res/`. However, since your issue is different from this one, I suggest opening a new issue and/or posting your issue on StackOverflow 😉

**Comment by Linaks** [comment]
Make this to your path_provider_app.xml:

`<?xml version="1.0" encoding="utf-8"?>
<paths xmlns:android="http://schemas.android.com/apk/res/android">
    <cache-path name="image" path="/"/>
    <external-path
        name="external_files"
        path="." />
</paths>`

**Comment by jakobheb** [comment]
I implemented it, but still not working. Do you have another idea why it's not working, I really tried everything.

**Comment by jakobheb** [comment]
When I use shareToInstagramFeed, I receive the error 
` android.content.ActivityNotFoundException: No Activity found to handle Intent { act=android.intent.action.SEND flg=0x10000001 pkg=com.instagram.android clip={null {T(0) U(content)}} (has extras) }
W/System.err( 9841): 	at android.app.Instrumentation.checkStartActivityResult(Instrumentation.java:2174)
W/System.err( 9841): 	at android.app.Instrumentation.execStartActivity(Instrumentation.java:1805)
W/System.err( 9841): 	at android.app.ContextImpl.startActivity(ContextImpl.java:1145)
W/System.err( 9841): 	at android.app.ContextImpl.startActivity(ContextImpl.java:1116)
W/System.err( 9841): 	at android.content.ContextWrapper.startActivity(ContextWrapper.java:441)
W/System.err( 9841): 	at com.appinio.socialshare.appinio_social_share.utils.SocialShareUtil.shareFileAndTextToPackage(SocialShareUtil.java:296)
W/System.err( 9841): 	at com.appinio.socialshare.appinio_social_share.utils.SocialShareUtil.shareToInstagramFeed(SocialShareUtil.java:70)
W/System.err( 9841): 	at com.appinio.socialshare.appinio_social_share.AppinioSocialSharePlugin.decideApp(AppinioSocialSharePlugin.java:89)
W/System.err( 9841): 	at com.appinio.socialshare.appinio_social_share.AppinioSocialSharePlugin.onMethodCall(AppinioSocialSharePlugin.java:59)
W/System.err( 9841): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:258)
W/System.err( 9841): 	at io.flutter.embedding.engine.dart.DartMessenger.invokeHandler(DartMessenger.java:295)
W/System.err( 9841): 	at io.flutter.embedding.engine.dart.DartMessenger.lambda$dispatchMessageToQueue$0$io-flutter-embedding-engine-dart-DartMessenger(DartMessenger.java:322)
W/System.err( 9841): 	at io.flutter.embedding.engine.dart.DartMessenger$$ExternalSyntheticLambda0.run(Unknown Source:12)
W/System.err( 9841): 	at android.os.Handler.handleCallback(Handler.java:942)
W/System.err( 9841): 	at android.os.Handler.dispatchMessage(Handler.java:99)
W/System.err( 9841): 	at android.os.Looper.loopOnce(Looper.java:226)
W/System.err( 9841): 	at android.os.Looper.loop(Looper.java:313)
W/System.err( 9841): 	at android.app.ActivityThread.main(ActivityThread.java:8757)
W/System.err( 9841): 	at java.lang.reflect.Method.invoke(Native Method)
W/System.err( 9841): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
W/System.err( 9841): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
I/ViewRootImpl@16d875f[MainActivity]( 9841): Resizing android.view.ViewRootImpl@45fd4d6: frame = [0,88][1080,2400] reportDraw = false forceLayout = false syncSeqId = 1
I/ViewRootImpl@16d875f[MainActivity]( 9841): handleResized, msg = 4 frames=ClientWindowFrames{frame=[0,88][1080,2400] display=[0,88][1080,2400] parentFrame=[0,0][0,0] parentClippedByDisplayCutout=false} forceNextWindowRelayout=false displayId=0 resizeMode=-1 frameChanged=false displayFrameChanged=false configChanged=false displayChanged=false`

**Comment by jakobheb** [comment]
Thank you for your help, I found the problem:)
Reason for not opening was compressing the image which worked in iOS, but not Android:)

**Comment by khanmujeeb687** [comment]
Hey @sadatd, Is your issue solved? have added the facebook app id?

**Comment by khanmujeeb687** [comment]
Hey @sadatd, I just tested and it is working fine for me. The sticker and background image both work perfectly in android and ios.

**Comment by sadatd** [comment]
Hey @khanmujeeb687, 
iOS works for me as well, only Android story is problem. It worked for me as well, but in last month or so it stopped working. Maybe it has to do something with instagram's update. Are you using latest version of Instagram?

**Comment by khanmujeeb687** [comment]
Hey @sadatd , I just tested with the latest Android version and its working perfectly fine, Facebook made it compulsory to add facebookId while sharing to insta story. But we have included it in the version 0.2.5.

**Comment by sadatd** [comment]
Hello @khanmujeeb687,
I appreciate your response. I initially attempted to resolve the issue with version 0.2.5, but unfortunately, I encountered the same error. However, after updating my Instagram app to the latest version, the problem magically disappeared. As it turns out, the issue was on Instagram's side, as I had mentioned in my previous comment. So, I'm happy to report that this issue can now be considered closed! 😄

I'd also like to express my gratitude for your package – it's greatly appreciated!
